site/src/Librecores/ProjectRepoBundle/Repository/CommitRepository.php

```diff
@@ -189,31 +195,42 @@ private function getCommitHistogramByWeek(
             ->getQuery()
             ->getResult('group');
 
-        // insert missing values
+        if (!empty($result)) {
+            // insert missing values
+            $startYear = array_keys($result)[0];
+            $startWeek = array_keys($result[$startYear])[0];
 
-        list($startYear, $startWeek) = explode(',', $start->format('Y,W'));
-        list($endYear, $endWeek) = explode(',', $end->format('Y,W'));
+            list($endYear, $endWeek) = explode(',', date('Y,W'));
 
-        for ($year = $startYear; $year <= $endYear; $year++) {
-            if (!array_key_exists($year, $result)) {
-                $result[$year] = [];
+            if (null !== $start) {
+                list($startYear, $startWeek) = explode(',', $start->format('Y,W'));
             }
 
-            // we use ==  because our keys are int but
-            // $startYear and $endYear are string
-            $week = $startYear == $year ? $startWeek : 1;
-            $weeklast = $endYear == $year ? $endWeek : \DateTimeImmutable::createFromFormat('d m Y',
-                "31 12 $year")->format('W');
+            if (null !== $end) {
+                list($endYear, $endWeek) = explode(',', $end->format('Y,W'));
+            }
 
-            for (; $week <= $weeklast; $week++) {
-                if (!array_key_exists($week, $result[$year])) {
-                    $result[$year][$week] = [0];    // fill non-existent values with zero
+            for ($year = $startYear; $year <= $endYear; $year++) {
+                if (!array_key_exists($year, $result)) {
+                    $result[$year] = [];
                 }
+
+                // we use ==  because our keys are int but
+                // $startYear and $endYear are string
```

Comment:
In this case you can make things more explicit without a comment by adding an `(int)` cast